src/plugins/console/server/routes/api/console/mappings/register_get_route.ts

```diff
@@ -0,0 +1,110 @@
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { IScopedClusterClient } from '@kbn/core/server';
+import type {
+  ClusterGetComponentTemplateResponse,
+  IndicesGetIndexTemplateResponse,
+  IndicesGetTemplateResponse,
+} from '@elastic/elasticsearch/lib/api/types';
+import { parse } from 'query-string';
+import type { RouteDependencies } from '../../..';
+import { API_BASE_PATH } from '../../../../../common/constants';
+
+interface Settings {
+  indices: boolean;
+  fields: boolean;
+  templates: boolean;
+  dataStreams: boolean;
+}
+
+async function retrieveSettings(
+  esClient: IScopedClusterClient,
+  settingsKey: keyof Settings,
+  settings: Settings
+) {
+  // Fetch autocomplete info if setting is set to true, and if user has made changes.
+  if (settings[settingsKey]) {
+    switch (settingsKey) {
+      case 'indices':
+        return esClient.asInternalUser.indices.getAlias();
+      case 'fields':
+        return esClient.asInternalUser.indices.getMapping();
+      case 'dataStreams':
+        return esClient.asInternalUser.indices.getDataStream();
+      case 'templates':
+        return Promise.all([
+          esClient.asInternalUser.indices.getTemplate(),
+          esClient.asInternalUser.indices.getIndexTemplate(),
+          esClient.asInternalUser.cluster.getComponentTemplate(),
+        ]);
+      default:
+        return Promise.resolve({});
+    }
+  } else {
+    // If the user doesn't want autocomplete suggestions, then clear any that exist
+    return Promise.resolve({});
+  }
+}
+
+async function getMappings(esClient: IScopedClusterClient, settings: Settings) {
+  return await retrieveSettings(esClient, 'fields', settings);
+}
+
+async function getAliases(esClient: IScopedClusterClient, settings: Settings) {
+  return await retrieveSettings(esClient, 'indices', settings);
+}
+
+async function getDataStreams(esClient: IScopedClusterClient, settings: Settings) {
+  return await retrieveSettings(esClient, 'dataStreams', settings);
+}
+
+async function getTemplates(esClient: IScopedClusterClient, settings: Settings) {
+  return await retrieveSettings(esClient, 'templates', settings);
+}
+
+export function registerGetRoute({ router }: RouteDependencies) {
+  router.get(
+    {
+      path: `${API_BASE_PATH}/mappings`,
+      validate: false,
+    },
+    async (ctx, request, response) => {
+      try {
+        const settings = parse(request.url.search, { parseBooleans: true }) as unknown as Settings;
+        const esClient = (await ctx.core).elasticsearch.client;
+
+        const mappings = await getMappings(esClient, settings);
+        const aliases = await getAliases(esClient, settings);
+        const dataStreams = await getDataStreams(esClient, settings);
+        const [legacyTemplates, indexTemplates, componentTemplates] = (await getTemplates(
```

Comment:
If the user deselects "Templates" in the Console Settings modal, this line causes a 500 error because `retrieveSettings` will resolve to an empty object and this line expects an array.

We can fix this by replacing lines 18-68 with this:

```ts
async function getMappings(esClient: IScopedClusterClient, settings: Settings) {
  if (settings['fields']) {
    return esClient.asInternalUser.indices.getMapping();
  }
  // If the user doesn't want autocomplete suggestions, then clear any that exist.
  return Promise.resolve({});
}

async function getAliases(esClient: IScopedClusterClient, settings: Settings) {
  if (settings['indices']) {
    return esClient.asInternalUser.indices.getAlias();
  }
  // If the user doesn't want autocomplete suggestions, then clear any that exist.
  return Promise.resolve({});
}

async function getDataStreams(esClient: IScopedClusterClient, settings: Settings) {
  if (settings['dataStreams']) {
    return esClient.asInternalUser.indices.getDataStream();
  }
  // If the user doesn't want autocomplete suggestions, then clear any that exist.
  return Promise.resolve({});
}

async function getTemplates(esClient: IScopedClusterClient, settings: Settings) {
  if (settings['templates']) {
    return Promise.all([
      esClient.asInternalUser.indices.getTemplate(),
      esClient.asInternalUser.indices.getIndexTemplate(),
      esClient.asInternalUser.cluster.getComponentTemplate(),
    ]);
  }
  // If the user doesn't want autocomplete suggestions, then clear any that exist.
  return Promise.resolve([]);
}
```